config/rbac/service_account_token.yaml

```diff
@@ -0,0 +1,7 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  name: k8ssandra-operator-token
+  annotations:
+    kubernetes.io/service-account.name: "k8ssandra-operator"
+type: kubernetes.io/service-account-token
```

Comment:
This solves the issue with Kind 1.24+ images. From the [Kubernetes 1.24 upgrade notes](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.24.md#urgent-upgrade-notes):
> - The `LegacyServiceAccountTokenNoAutoGeneration` feature gate is beta, and enabled by default. When enabled, Secret API objects containing service account tokens are no longer auto-generated for every ServiceAccount. Use the [TokenRequest](https://kubernetes.io/docs/reference/kubernetes-api/authentication-resources/token-request-v1/) API to acquire service account tokens, or if a non-expiring token is required, create a Secret API object for the token controller to populate with a service account token by following this [guide](https://kubernetes.io/docs/concepts/configuration/secret/#service-account-token-secrets).

I used the second option which is functionally equivalent to what we had before. But maybe we should investigate how `TokenRequest` could be used.

Reply:
Excellent. Yes, we should indeed move to a newer API at some point (I didn't check the version ranges - but those are always problematic in Kubernetes). If you wish to create tickets, that would be nice.